src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Logger.cs

```diff
@@ -29,32 +32,40 @@ public class Logger
         private readonly HashSet<string> _trimWarnedAssemblies = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private readonly HashSet<string> _aotWarnedAssemblies = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
-        public static Logger Null = new Logger(new TextLogWriter(TextWriter.Null), false);
+        public static Logger Null = new Logger(new TextLogWriter(TextWriter.Null), null, false);
 
         public bool IsVerbose { get; }
 
-        public Logger(ILogWriter writer, bool isVerbose, IEnumerable<int> suppressedWarnings, bool singleWarn, IEnumerable<string> singleWarnEnabledModules, IEnumerable<string> singleWarnDisabledModules)
+        public Logger(
+            ILogWriter writer,
+            CompilerGeneratedState compilerGeneratedState,
```

Comment:
I had to keep the fact that `_compilerGeneratedState` can be null unfortunately - because the `Null` logger doesn't have `ILProvider` and the `CompilerGeneratedState` can't operate without one really.
Alternative would be to have an interface to the compiler generated state functionality needed by the logger and have a dummy implementation. Since the null only adds two places where it needs a null check it didn't feel that bad.